src/ToEnum.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter;
+
+use BackedEnum;
+use UnitEnum;
+
+use function assert;
+use function constant;
+use function defined;
+use function is_a;
+use function is_int;
+use function is_string;
+
+/**
+ * @psalm-type Options = array{
+ *     enum: class-string<UnitEnum>|class-string<BackedEnum>,
+ * }
+ * @implements FilterInterface<UnitEnum|BackedEnum>
+ */
+final class ToEnum implements FilterInterface
+{
+    /** @var class-string<UnitEnum>|class-string<BackedEnum> */
+    private readonly string $enum;
+
+    /** @param Options $options */
+    public function __construct(array $options)
+    {
+        $this->enum = $options['enum'];
```

Comment:
No check for the array key?

Reply:
Theoretically static analysis has our back here because the `Options` type is `array{enum: ...}`.

I can change this to be ultra defensive and validate the option is set, and, `class-string<UnitEnum>|class-string<BackedEnum>`.

Reply:
My problem is much simpler:

```php
$filter = new Laminas\Filter\ToEnum();
```

How does this end or what does it lead to?

Reply:
This is a `TypeError` assigning null to typed property string, so a hard crash.

Reply:
Preceded by a warning also for the undefined array key

Reply:
I am sorry, I ca not remember, but is that what we want? Or rather throw an exception?
It seems that is what we want:

https://github.com/laminas/laminas-filter/blob/3d0607220564380a6276eda3078f64948760e8f1/src/DecompressArchive.php#L34-L39

I am thinking about how we should document this.

Reply:
Very good point.

In this case, if we add defensive checks, Psalm will moan at us about the redundant checks with good reason, because the required type is precisely defined.

In the case you quoted above, Psalm cannot tell the user that an arbitrary string is not a directory, so the defensive check is required.

Overall, the defensive checks provide a better user experience for those not using static analysis by way of clear exception messages and traces.

In the case of filter, SA is a bit of a moot point because inference will not occur when filters are thrown together from an array input-filter specification.

Not sure what the best approach is.

Maybe @Ocramius or @weierophinney could chime in with opinions on best-practice here.